### PR TITLE
[CARBONDATA-3814]Remove dead code and refactor MV events

### PR DESCRIPTION
**Pull request by akashrn5**
### Why is this PR needed?
 Some code is not used in Index Events, some MV events should be at one place
 
 ### What changes were proposed in this PR?
Removed the dead code and refactored the existing MV events class
    
 ### Does this PR introduce any user interface change?
 - No

 ### Is any new testcase added?
 - No

### CODE REVIEW

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2972/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1254/

File: integration/spark/src/main/scala/org/apache/carbondata/view/MVEvents.scala [hunk 1]
```diff
@@ -20,31 +20,43 @@ package org.apache.carbondata.view
...
 
 /**
- * For handling operation's before start of mv creation
+ * For handling operation's before start of MV creation
  */
-case class CreateMVPreExecutionEvent(sparkSession: SparkSession,
-    tableIdentifier: TableIdentifier) extends Event
+case class CreateMVPreExecutionEvent(
+    sparkSession: SparkSession,
+    storePath: String,
```
Comment (Indhumathi27) [inline integration/spark/src/main/scala/org/apache/carbondata/view/MVEvents.scala:30]: Can rename `storePath` to `systemDirectoryPath` in all mv events

File: integration/spark/src/main/scala/org/apache/spark/sql/execution/command/view/CarbonRefreshMVCommand.scala [hunk 2]
```diff
@@ -33,33 +33,33 @@ import org.apache.carbondata.view.{MVManagerInSpark, MVRefresher, RefreshMVPostE
...
-    val identifier = TableIdentifier(name, Option(databaseName))
+    val identifier = TableIdentifier(mvName, Option(databaseName))
     val operationContext = new OperationContext()
     OperationListenerBus.getInstance().fireEvent(
       RefreshMVPreExecutionEvent(session, identifier),
       operationContext)
     viewManager.setStatus(schema.getIdentifier, MVStatus.ENABLED)
     OperationListenerBus.getInstance().fireEvent(
-      RefreshMVPostExecutionEvent(session, identifier),
+      RefreshMVPostExecutionEvent(session, identifier, mvName),
```
Comment (Indhumathi27) [inline integration/spark/src/main/scala/org/apache/spark/sql/execution/command/view/CarbonRefreshMVCommand.scala:62]: Please check if  `RefreshMVPreExecutionEvent` is used. If not used, please remove

File: integration/spark/src/main/scala/org/apache/carbondata/view/MVEvents.scala (hunk 1 above)
Comment (akashrn5) [reply integration/spark/src/main/scala/org/apache/carbondata/view/MVEvents.scala]: done

File: integration/spark/src/main/scala/org/apache/spark/sql/execution/command/view/CarbonRefreshMVCommand.scala (hunk 2 above)
Comment (akashrn5) [reply integration/spark/src/main/scala/org/apache/spark/sql/execution/command/view/CarbonRefreshMVCommand.scala:62]: its being used

Comment (Indhumathi27) [inline integration/spark/src/main/scala/org/apache/spark/sql/execution/command/view/CarbonRefreshMVCommand.scala:62]: mvName is present in TableIdentifier itself. Can remove mvName from RefreshMVPostExecutionEvent

Comment (akashrn5) [reply integration/spark/src/main/scala/org/apache/spark/sql/execution/command/view/CarbonRefreshMVCommand.scala]: changed

File: integration/spark/src/main/scala/org/apache/carbondata/events/Events.scala [hunk 3]
```diff
@@ -195,7 +195,7 @@ trait CreateCarbonRelationEventInfo {
  */
 trait MVEventsInfo {
   val sparkSession: SparkSession
-  val storePath: String
+  val systemDirectoryPath: String
```
Comment (niuge01) [inline integration/spark/src/main/scala/org/apache/carbondata/events/Events.scala:198]: The field name changed, but i can not found any code which used the field has be changed, it's will be OK？

Comment (akashrn5) [reply integration/spark/src/main/scala/org/apache/carbondata/events/Events.scala:198]: yes, this is just the event, if any user wants to implement a listener for this event, then he will use these variable. So it is ok

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2988/

**Comment by akashrn5** [comment]
retest this please

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2990/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1271/

**Comment by kunal642** [comment]
LGTM

**Pull request closed without merging**